src/python/module/smartredis/util.py

```diff
@@ -111,3 +114,93 @@ def typecheck(arg, name, _type):
     """
     if not isinstance(arg, _type):
         raise TypeError(f"Argument {name} is of type {type(arg)}, not {_type}")
+
+
+
+class DatasetConverter:   
+    
+    def add_metadata_for_xarray(ds,data=None,dims=None,coords=None,attrs=None): #time=None,time_variable=None):
+        """_summary_
+        return attribute labels, dimensions labels, coordinate labels etc for retrieval 
+        adding the field names to retrieve to xarray specific field names 
+        can use those general field names to construct xarray in transform method 
+
+        Args:
+            ds (_type_): _description_
+            data (_type_, optional): _description_. Defaults to None.
+            dims (_type_, optional): _description_. Defaults to None.
+            coords (_type_, optional): _description_. Defaults to None.
+            attrs (_type_, optional): _description_. Defaults to None.
```

Comment:
While I appreciate the documentation, I don't think that this formatting is compatible with our doc generation. Please take a look at `src/python/module/client.py` for an example of what it needs to look like